lib/Convos/Core/Backend/File.pm

```diff
@@ -80,13 +80,13 @@ sub messages {
   $re = qr{\Q$re\E}i unless ref $re;
   $re = qr/^(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}) (.*$re.*)$/;
 
-  $args{after}  = Time::Piece->strptime($query->{after},  '%Y-%m-%dT%H:%M:%S')if $query->{after};
-  $args{before} = Time::Piece->strptime($query->{before}, '%Y-%m-%dT%H:%M:%S') if $query->{before};
-  # If neither before nor after are defined, search everything before now
-  $args{before} = gmtime if !$args{before} && !$args{after};
   $args{limit}    = $query->{limit} || 60;
   $args{messages} = [];
   $args{re}       = $re;
+  # TODO: What to do when after is after before?
+  $args{before}   = $query->{before} ? Time::Piece->strptime($query->{before}, '%Y-%m-%dT%H:%M:%S') : gmtime;
+  $args{after}    = $query->{after}  ? Time::Piece->strptime($query->{after},  '%Y-%m-%dT%H:%M:%S') : $args{before}->add_months(-1);
```

Comment:
That's a great question and because of the expense of opening all these files on disk it seems like restricting to a max of 12 months would be wise.  Should we make that a user-config option so that anyone controlling their own convos could override it?  I'm think about the {limit}, and how it has the ability to be overridden.  I think we should do it in the same way: hardcode to 12 but let {max_months} (or whatever) in the query override that hard-coded 12.

Or, I'm not sure where the query can get set.  Can, theoretically, the front end code control that attribute?  We wouldn't want someone to create a DoS thru a large {limit} and vast {max_months}.